### ISSUE TITLE
Amplify CLI provision for converting resolver to pipeline and chaining with other AppSync functions.

### ISSUE THREAD

**Issue by ramakrishnan** [aws-amplify/amplify-cli#5785]
**Is your feature request related to a problem? Please describe.**
Customising resolver pipelines and chaining with other AppSync functions through cli 

**Describe the solution you'd like**
`amplify api addFunction`
`amplify api convertToPipeline`

**Describe alternatives you've considered**
Currently there are few workaround in updating the `customResource.json` but the process is too cumbersome and needs heavy maintenance. 

**Additional context**
Using direct AWS API to create and manage. But the down side is on every amplify push, amplify wipes off the changes with what is present in customResource json.
Currently invoking the AWS API scripts after every amplify push.

**Comment by akshbhu** [comment]
Hi @ramakrishnan 

If I am understanding correctly we have @function directive that should support the use case you describe. The @function directive supports AWS AppSync pipeline resolvers (chaining functions) . Here is the documentation: https://docs.amplify.aws/cli/graphql-transformer/function#chaining-functions.

If this is not the case , Can you elaborate more on your use case?

**Comment by ramakrishnan** [comment]
@akshbhu 
Agree `@function` will internally create the pipeline but it is currently limited to `Query`, in order to get this working for a `Mutation` one has to create a custom mutation. please correct me if I am wrong.

To elaborate my use case.

I would like to use the existing CRUD functions created while using `@model` directive and chain additional custom function.

Example I would like to apply a custom pre-validation on every resolver (For all CRUD operations) in this case I would either create a AppSync function having data source as  Lambda or HTTP. I would now like to convert all the existing resolver into pipeline and chain this AppSync function.

Talking about `@function` it makes me ask of additional use case is it possible to specify the operations where the function has to be applied?
Example

```
// schema.graphql
type User @model @function(name: "customValidation") {
   id: ID!
  username: String!
  password: String! @function(name: "hashPassword", on: ["create"])
}
```
So that when anyone who looks at schema.graphql will understand about the model and the custom operation happening to password filed during create, without even going through the pipeline chain.

Basically my questions can be summarised into two.
1. How to include a pipeline function on a model level chaining to its existing CRUD functions
2. How to restrict function per operation, when added to an attribute

**Comment by akshbhu** [comment]
Hello @ramakrishnan 

I understood your point.Currently Amplify CLI does not currently support pipeline resolvers but this is something we are working on.Follow this RFC for those updates and some possible implementations: #1055
Closing in favour of #1055
